cmd/backup/main.go

```diff
@@ -82,6 +83,8 @@ func main() {
 
 	s.must(s.encryptBackup())
 	s.must(s.copyBackup())
+	s.stats.EndTime = time.Now()
```

Comment:
Just curious: why did you decide to define "end" _before_ pruning happens?

Reply:
Let's say that I have a big volume, like the Plex database, which consists of a lot of files, and is quite big (for example mine has 400k files and is 21 GiB). The "timeline" of actions, along with their duration, could look something like this:
```
20min:		tar the archive
5min:		encrypt the archive
1h:		upload to remote host
1h30min:	wait retention leeway period
10sec:		prune backups
10sec:		send notifications
```
The retention period babsically has to be the sum of previous operations, since the pruning algorithm checks for the last modified date on the files.
Now if someone asks me how much time it takes for the backup to run, I'd reply about 1 and a half hours, not 3 hours because of the delayed clean operation.


Now that I've written all this, it occured to me that there is a way simpler way of handling pruning: why not simply add to the deadline `BackupPruningLeeway` and get rid of the sleep. Wouldn't it achieve the same exact result? Or is there a particular reason for the sleep to be used?

The code would then simply be:
```go
deadline := time.Now().AddDate(0, 0, -int(s.c.BackupRetentionDays)).Add(s.c.BackupPruningLeeway)
```

Reply:
Thanks for explaining, that makes sense.

> The retention period babsically has to be the sum of previous operations, since the pruning algorithm checks for the last modified date on the files.

If I recall correctly, it should be the fluctuation between slow and fast backup runs, not the entire time span. The sleep is there to prevent the following from happening:
- you have a very slow backup run (your system is busy or whatever) that takes 6 minutes
- 7 days later the next backup run takes the expected 5 minutes and looks for backups to prune
- the slow backup is not considered as it's still inside of the 7 day retention period

A leeway of 2 minutes should solve this, there's no need to use 5. Thishas been introduced ages ago https://github.com/offen/docker-volume-backup/commit/2892369677b042262a13c52c3b8a1547bce94ebc when this was still a shell script.

> it occured to me that there is a way simpler way of handling pruning: why not simply add to the deadline BackupPruningLeeway and get rid of the sleep.

You are totally correct. I think I just blindly ported this from the shell version (where it had to be done like this as you wouldn't have fine grained control over the deadline as easily) and never thought about this option.

Do you think it makes sense to change this in the context of this PR? In case it does not make your life easier, it might be easier to do it in a separate step for bookkeeping and such.

Reply:
> If I recall correctly, it should be the fluctuation between slow and fast backup runs, not the entire time span.

Of course, you're right! Note to self: think twice when working late at night.

> Do you think it makes sense to change this in the context of this PR? 

While not entirely on scope for the PR, it's a very easy fix and would be in its own commit, so I don't think it's a problem doing it now. Let me know if I should proceed.

Reply:
Yeah, let's include. It makes a lot of sense and will save quite some time for some people.